src/integrationTest/java/com/mongodb/spark/sql/connector/mongodb/MongoSparkConnectorTestCase.java

```diff
@@ -140,6 +157,59 @@ public void loadSampleData(
     HELPER.loadSampleData(numberOfDocuments, sizeInMB, config);
   }
 
+  /** Runs events with the profiler on. */
+  public void assertCommentsInProfile(final Runnable runnable, final ReadConfig readConfig) {
+    assumeFalse(isSharded());
+    assertNotNull(readConfig.getComment());
+    assertCommentsInProfile(runnable, readConfig.getDatabaseName(), readConfig.getCollectionName());
+  }
+
+  /** Runs events with the profiler on. */
+  public void assertCommentsInProfile(final Runnable runnable, final WriteConfig writeConfig) {
+    assertNotNull(writeConfig.getComment());
+    assertCommentsInProfile(
+        runnable, writeConfig.getDatabaseName(), writeConfig.getCollectionName());
+  }
+
+  private void assertCommentsInProfile(
+      final Runnable runnable, final String databaseName, final String collectionName) {
+    assumeTrue(isAtLeastFiveDotZero());
+    assumeFalse(isSharded());
+    MongoDatabase database = HELPER.getMongoClient().getDatabase(databaseName);
+    MongoCollection<Document> profileCollection = database.getCollection("system.profile");
+    try {
+      profileCollection.drop();
+      database.runCommand(
+          BsonDocument.parse(
+              format("{profile: 2, filter: {ns: '%s.%s'}}", databaseName, collectionName)));
+
+      runnable.run();
+
+      Predicate<? super Document> commentsToIgnore =
+          d -> {
+            String profileComment = d.get("comment", "");
+            return d.containsKey("killCursors")
+                || profileComment.equals(IGNORE_COMMENT)
+                || profileComment.equals(TEST_COMMENT);
```

Comment:
Why are both "ignore" and "test" comments needed?

Reply:
Good question: as both are excluded from the profiler logs, in one sense only one could be required.

However, in the tests it makes sense to differentiate between the configured comments (eg `TEST_COMMENT`) and any setup or assertion based commands that also generate profile logs.

I've updated the test code to filter out explicitly ignored commands and the searching of any other commands that are missing a comment. Hopefully, that will help clarify it.

```
      List<Document> profileDocs =
          profileCollection
              .find(
                  Filters.nor(Filters.exists("killCursors"), Filters.eq("comment", IGNORE_COMMENT)))
              .into(new ArrayList<>());

      List<String> withoutComment =
          profileDocs.stream()
              .filter(d -> !d.getEmbedded(asList("command", "comment"), "").equals(TEST_COMMENT))
              .map(Document::toJson)
              .collect(Collectors.toList());
```